File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
       unordered_map<int , int > count;

       for(int i = 0 ;i < nums.size(); i++)
       {

            count[nums[i]]++;

       } 

       //2 -> move map to vector to sort 

       vector<pair < int , int >> freq(count.begin() , count.end());

       // sort 

       sort(freq.begin() , freq.end() , 
       [](const pair <int , int > &a , const pair<int , int > &b){
       return a.second > b .second ;
       });

        // push it final result ;

        vector<int> result;
        for(int i = 0 ; i < k ; i++)
        {
            result.push_back(freq[i].first);
        }

            return result;
    }
};

File: Data Structures & Algorithms/counting-bits/submission-0.java
class Solution {
    public int[] countBits(int n) {

     int []res = new int[n+1];

     while( n>=1)
     {  
        int num = n;
        int count = 0 ;
          while(num != 0)
        {
            count += num % 2 ;
            num = num >> 1;
        }

        res[n] = count;
        n--;

     }   
     return res;
    }
}
